### PR TITLE
ConfigurationGathering.py: Documents arg_parser

### PR DESCRIPTION
**Pull request by JohnS-01**
Documents arg_parse in load_configuration function. 

Closes https://github.com/coala/coala/issues/4101

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by JohnS-01** [comment]
Documented on ConfigurationGathering.py

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
     :param arg_list:    The list of command line arguments.
     :param log_printer: The LogPrinter object for logging.
+    :param arg_parser:  An ``ArgParser`` object. See functions - get_filtered_bears(), gather_configuration()
```
Comment (Udayan12167) [inline coalib/settings/ConfigurationGathering.py]: Don't use see another method. Just copy the definition here as well

**Comment by supergr35** [comment]
Couple things :)
Squash the commits
Fix the gitmate errors (note: you can run coala on your code locally to help you do this!)

**Comment by JohnS-01** [comment]
How do I squash commits through the browser?

File: coalib/settings/ConfigurationGathering.py [hunk 2]
```diff
@@ -143,6 +143,7 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
     :param arg_list:    The list of command line arguments.
     :param log_printer: The LogPrinter object for logging.
+    :param arg_parser:  An ``ArgParser`` object. 
     :return:            A tuple holding (log_printer: LogPrinter, sections:
```
Comment (JohnS-01) [inline coalib/settings/ConfigurationGathering.py]: Why is it saying I have trailing white spaces? I fixed it?

**Comment by supergr35** [comment]
I am really not sure how to do it through browser. Do you not have terminal access?

File: coalib/settings/ConfigurationGathering.py [hunk 3]
```diff
@@ -143,6 +143,7 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
     :param arg_list:    The list of command line arguments.
     :param log_printer: The LogPrinter object for logging.
+    :param arg_parser:  An ``ArgParser`` object.
```
Comment (userzimmermann) [inline coalib/settings/ConfigurationGathering.py]: Should be `ArgumentParser` or fully qualified `argparse.ArgumentParser` to clearly know what is meant here. I know that just `ArgParse` is used several times in `coalib` doc strings... I will open an issue for that

Comment (jayvdb) [inline coalib/settings/ConfigurationGathering.py]: Yes, John, please fix just this one line as @userzimmermann has requested.
Someone else will fix all the others.

**Comment by userzimmermann** [comment]
@JohnS-01 Need further help with rebasing? There are some useful resources regarding this:

http://api.coala.io/en/latest/Developers/Git_Basics.html

https://help.github.com/articles/about-pull-request-merges/

**Comment by JohnS-01** [comment]
I'll do it today, but I'll need some time to squash/rebash my commits. I've edited the .py file through the browser as I will have to fix authentication stuff with my Git bash terminal.

**Comment by userzimmermann** [comment]
Hey @JohnS-01 :) If you don't feel so comfortable on the command line, you should take a look at https://www.gitkraken.com

It's IMO the fanciest Git GUI out there, also making squashing and rebasing very simple: https://support.gitkraken.com/working-with-files/commits

**Comment by JohnS-01** [comment]
@userzimmermann Thanks I've downloaded GitKraken. I'm still having the same problems though. GitKraken won't let me clone anything.

**Comment by userzimmermann** [comment]
@JohnS-01 You first have to select all the commits of your branch you want to squash

Look again at the squashing video in https://support.gitkraken.com/working-with-files/commits

**Comment by JohnS-01** [comment]
@ userzimmermann  I though I did select all the commits (see image).

**Pull request closed without merging**